src/library-authoring/LibraryAuthoringPage.tsx

```diff
@@ -191,33 +231,14 @@ const LibraryAuthoringPage = () => {
               <Tab eventKey={TabList.components} title={intl.formatMessage(messages.componentsTab)} />
               <Tab eventKey={TabList.collections} title={intl.formatMessage(messages.collectionsTab)} />
             </Tabs>
-            <Routes>
-              <Route
-                path={TabList.home}
-                element={(
-                  <LibraryHome tabList={TabList} handleTabChange={handleTabChange} />
-                )}
-              />
-              <Route
-                path={TabList.components}
-                element={<LibraryComponents variant="full" />}
-              />
-              <Route
-                path={TabList.collections}
-                element={<LibraryCollections variant="full" />}
-              />
-              <Route
-                path="*"
-                element={<NotFoundAlert />}
-              />
-            </Routes>
```

Comment:
We had to change to render based on a state instead of the route because of the `ComponentPicker` component

Reply:
Can you elaborate? Why can't we have nested routes, with three home/components/collections routes underneath /library/foo and the same three home/components/collections routes also available underneath /library-component-picker/ ?

Reply:
Hm.. You're right. We actually can. I didn't think this way when I refactored this.
Do you prefer we stick to the routing approach?

Reply:
Well what I really want is to implement https://github.com/openedx/frontend-app-authoring/issues/1230 ASAP, which will simplify everything. Then the "All/Collections/Components" tabs will essentially be just another filter. But yeah, I do think it should be in the route if possible.

Reply:
Though it doesn't need to be in the route for the content picker.

Reply:
Just to be clear, it is still in the URL path as it was. I'm just using states from the context to render the child components instead of the `Router` component.

Reply:
Oh ok. Maybe that's not so bad? I'll have a look later.

Reply:
I think it would be better to fix the `Route` issue with #1230. I moved more things to the context here, which will help the refactor a bit.

This PR is "functional".
We will probably want to change the endpoint to add the content, but is it possible to test it? It is missing the Modal tough.
I will post a more detailed update about the task status tomorrow.

Reply:
This change seems to have broken the Back/Forward button in the browser. Before, you could change tabs and then press BACK and it would return to the previous tab, but now it's not working.